src/chemnlp/data_val/config.py

```diff
@@ -58,3 +58,9 @@ class TrainPipelineConfig(BaseModel):
     prompt_tuning: PromptTune
     trainer: TrainerConfig
     wandb: WandbConfig
+
+
+class DataMixingConfig(BaseModel):
```

Comment:
For the `data_paths` and `data_proportions` could we add the element data type in each list? This helps new people who don't have an example configuration file understand the types without looking at the code. 
i.e.
```python
data_paths: list -> data_paths: list[str]
```

Also, I know this way of writing types as `list` instead of `typing.List` is new in Python 3.10 so [it's not compatible with 3.8](https://adamj.eu/tech/2022/10/17/python-type-hints-old-and-new-syntaxes/). However, version 3.8 is allowed as per our [pyproject.toml](https://github.com/OpenBioML/chemnlp/blob/main/pyproject.toml#L9). Even though I know we use 3.10 on the cluster, other files use `typing` across the repository so it might be better for compatibility to change these to `typing.List`?